src/compiler/build/build-stats.ts

```diff
@@ -71,36 +71,39 @@ export function generateBuildStats(
 
 /**
  *
- * @param config
- * @param buildCtx
+ * @param config the project build configuration
+ * @param buildCtx An instance of the build which holds the details about the build
  * @returns
  */
 export async function writeBuildStats(config: d.Config, data: d.CompilerBuildStats | { diagnostics: d.Diagnostic[] }) {
-  const statsTargets = config.outputTargets.filter((o) => o.type === 'stats') as d.OutputTargetStats[];
+  const statsTargets = config.outputTargets.filter(isOutputTargetStats);
 
   await Promise.all(
-    statsTargets.map((outputTarget) => {
-      config.sys.writeFile(outputTarget.file, JSON.stringify(data, null, 2));
+    statsTargets.map(async (outputTarget) => {
+      const result = await config.sys.writeFile(outputTarget.file, JSON.stringify(data, null, 2));
+
+      if (result.error) {
+        config.logger.warn([`Stats failed to write file to ${outputTarget.file}: ${result.error}`]);
```

Comment:
Have we verified this won't print 
```
Stats failed to write file to [Object object]: [Object object]
```
?

Reply:
file should work fine, I removed the result.error so we can guard against that.